src/storage/SparqlResourceStore.ts

```diff
@@ -0,0 +1,753 @@
+        } as GraphPattern,
+      ],
+      type: 'query',
+      prefixes: {},
+    } as ConstructQuery;
+    return (await this.sendSparqlQuery(representationQuery)).results.bindings;
+  }
+
+  /**
+   * Helper function to get the representation of a document resource.
+   * @param resourceURI - Identifier of the resource to retrieve.
+   */
+  private async getResourceRepresentation(resourceURI: string): Promise<Representation> {
+    // Get the triples from the resourceURI graph and from the corresponding metadata graph.
+    const data: Quad[] = await this.getSparqlRepresentation(resourceURI);
+    const metadata: Quad[] = await this.getSparqlRepresentation(`${resourceURI}.metadata`);
+
+    // Only include the triples of the resource graph in the data readable.
+    const readableData = streamifyArray([ ...data ]);
+    return {
+      dataType: DATA_TYPE_QUAD,
+      data: readableData,
+      metadata: {
+        raw: metadata,
+        contentType: CONTENT_TYPE_QUADS,
+      },
+    };
+  }
+
+  /**
+   * Helper function to get the representation of a container.
+   * @param containerURI - Identifier of the container to retrieve.
+   */
+  private async getContainerRepresentation(containerURI: string): Promise<Representation> {
+    // Get the triples from the containerURI graph and from the corresponding metadata graph.
+    const data: Quad[] = await this.getSparqlRepresentation(containerURI);
+    const metadata: Quad[] = await this.getSparqlRepresentation(`${containerURI}.metadata`);
+
+    // Include both the triples of the resource graph and the metadata graph in the data readable.
+    const readableData = streamifyArray([ ...data, ...metadata ]);
+    return {
+      dataType: DATA_TYPE_QUAD,
+      data: readableData,
+      metadata: {
+        raw: metadata,
+        contentType: CONTENT_TYPE_QUADS,
+      },
+    };
+  }
+
+  /**
+   * Helper function to make sure that all incoming quads are in the default graph.
+   * If the incoming quads are undefined, an empty array is returned instead.
+   * @param type - Type of the quads to indicate in the possible error.
+   * @param quads - Incoming quads.
+   *
+   * @throws {@link ConflictHttpError}
+   * If one or more quads are not in the default graph.
+   */
+  private ensureValidQuads(type: string, quads?: Quad[]): Quad[] {
+    if (quads) {
+      if (!quads.every((x): any => inDefaultGraph(x))) {
+        throw new ConflictHttpError(`All ${type} quads should be in the default graph.`);
+      }
+      return quads;
+    }
+    return [];
+  }
+
+  /**
+   * Helper function to get the parent container URI of a container URI.
+   * @param containerURI - Incoming container URI.
+   */
+  private getParentContainer(containerURI: string): string {
+    const [ , parentContainerURI ] = /^(.*\/)[^/]+\/$/u.exec(containerURI) ?? [];
+    if (typeof parentContainerURI !== 'string') {
+      throw new Error('Invalid containerURI passed.');
+    }
+    return parentContainerURI;
+  }
```

Comment:
This actually already exists in the `src/storage/ContainerManager.ts` (specific implementation `UrlContainerManager`), so you can take that class as constructor argument.